isis/tests/FunctionalTestsMvic2isis.cpp

```diff
@@ -0,0 +1,489 @@
+#include <QTemporaryDir>
+
+#include "mvic2isis.h"
+#include "Fixtures.h"
+#include "Pvl.h"
+#include "PvlGroup.h"
+#include "TestUtilities.h"
+#include "Histogram.h"
+
+#include "gtest/gtest.h"
+#include "gmock/gmock.h"
+
+#include "Fixtures.h"
+
+using namespace Isis;
+
+static QString APP_XML = FileName("$ISISROOT/bin/xml/mvic2isis.xml").expanded();
+
+TEST_F(TempTestingFiles, Mvic2IsisTestTdiMode) {
+  Pvl appLog;
+  QString cubeFileName = tempDir.path() + "/mvic2isisTEMP.cub";
+  QString undistortedCubeName = tempDir.path() + "mvic2isisUndistorted.cub";
+  QString errorCubeName = tempDir.path() + "mvic2isisError.cub";
+  QString qualityCubeName = tempDir.path() + "mvic2isisQuality.cub";
+  QVector<QString> args = {"from=data/mvic2isis/mc3_0034948318_0x536_sci_1_cropped.fits",
+                           "to=" + cubeFileName,
+                           "undistorted="+undistortedCubeName,
+                           "error="+errorCubeName,
+                           "quality="+qualityCubeName};
+
+  UserInterface options(APP_XML, args);
+  try {
+   mvic2isis(options, &appLog);
+  }
+  catch (IException &e) {
+    FAIL() << "Unable to ingest MVIC image: " << e.toString().toStdString().c_str() << std::endl;
+  }
+  Cube cube(cubeFileName);
+  Pvl *isisLabel = cube.label();
+
+  // Dimensions Group
+  PvlGroup &dimensions = isisLabel->findGroup("Dimensions", Pvl::Traverse);
+  ASSERT_EQ(int(dimensions["Samples"]), 1);
+  ASSERT_EQ(int(dimensions["Lines"]), 3);
+  ASSERT_EQ(int(dimensions["Bands"]), 1);
```

Comment:
no point in changing it now, but in the future, cube->samplecount/linecount/bandcount would be better.